core/src/main/java/com/linecorp/armeria/common/HttpHeadersBase.java

```diff
@@ -470,6 +474,99 @@ public long contentLength() {
         }
     }
 
+    List<MediaType> accept() {
+        @SuppressWarnings("unchecked")
+        final List<MediaType> cached = (List<MediaType>) cache.get(HttpHeaderNames.ACCEPT);
+        if (cached != null) {
+            if (cached instanceof ImmutableList) {
+                return cached;
+            } else {
+                // Should return an immutable list.
+                final List<MediaType> immutableCache = ImmutableList.copyOf(cached);
+                cache.put(HttpHeaderNames.ACCEPT, immutableCache);
+                return immutableCache;
+            }
+        }
+
+        final List<String> acceptHeaders = getAll(HttpHeaderNames.ACCEPT);
+        if (acceptHeaders.isEmpty()) {
+            cache.put(HttpHeaderNames.ACCEPT, ImmutableList.of());
+            return ImmutableList.of();
+        }
+
+        // Cache miss. Parse Accept headers and store it to cache.
+        final List<MediaType> acceptTypes = new ArrayList<>(4);
+        for (final String acceptHeader : acceptHeaders) {
+            for (String accept : ACCEPT_SPLITTER.split(acceptHeader)) {
+                try {
+                    acceptTypes.add(MediaType.parse(accept));
+                } catch (IllegalArgumentException ignored) {
+                    // Ignore a malformed MediaType.
+                }
+            }
+        }
+
+        if (acceptTypes.size() > 1) {
+            acceptTypes.sort(HttpHeadersBase::compareMediaType);
+        }
+        final List<MediaType> parsed = ImmutableList.copyOf(acceptTypes);
+        cache.put(HttpHeaderNames.ACCEPT, parsed);
+        return parsed;
+    }
+
+    final void accept(Iterable<MediaType> newAcceptTypes) {
+        requireNonNull(newAcceptTypes, "newAcceptTypes");
+        final int size = Iterables.size(newAcceptTypes);
+        checkArgument(size > 0, "newAcceptTypes is empty");
+
+        @SuppressWarnings("unchecked")
+        List<MediaType> cachedAcceptTypes = (List<MediaType>) cache.get(HttpHeaderNames.ACCEPT);
+        if (cachedAcceptTypes == null) {
+            // Set new Accept headers
+            final List<MediaType> copied = new ArrayList<>(size + 2);
```

Comment:
What is the `+2` in this case?

Reply:
Rule of thumbs just in case users add additional accept headers. 😆
I guess most users will set one accept header. At most, it will be two or three.

Reply:
I never thought that this could be called multiple times. SGTM.